### PR TITLE
Update readme to use API token

### LINKED ISSUE #6: Implement user/pass authentication to argocd
### What problem are you facing?
Currently `provider-argocd` uses a JWT to authenticate to argocd. This JWT is manually created for the argocd admin user: 
```bash
# grab the initial admin password
ARGOCD_SECRET=$(kubectl view-secret argocd-initial-admin-secret -n argocd -q)
# port forward the argocd api to the host:
kubectl -n argocd port-forward svc/argocd-server 8443:443
# create a JWT for the admin user at the argocd api
ARGOCD_TOKEN=$(curl -s -X POST -k -H "Content-Type: application/json" --data '{"username":"admin","password":"'$ARGOCD_SECRET'"}' https://localhost:8443/api/v1/session | jq -r .token)
```
The JWT is time-bound and needs to be recreated every X hours.

### How could Crossplane help solve your problem?
Instead of accepting a JWT to connect to argocd, `provider-argocd` should accept a user/pass and (re-)create the JWT on its own.

### PR DESCRIPTION
**Pull request by MisterMX**
### Description of your changes

Fixes #6 by changing the readme to use API instead of session tokens which do not expire.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,8 +28,12 @@ Install the [kubectl-view-secret](https://github.com/elsesiy/kubectl-view-secret
...
 # port forward the argocd api to the host:
 kubectl -n argocd port-forward svc/argocd-server 8443:443
-# create a JWT for the admin user at the argocd api
-ARGOCD_TOKEN=$(curl -s -X POST -k -H "Content-Type: application/json" --data '{"username":"admin","password":"'$ARGOCD_SECRET'"}' https://localhost:8443/api/v1/session | jq -r .token)
+# create a session JWT for the admin user at the argocd api
+ARGOCD_SESSION_TOKEN=$(curl -s -X POST -k -H "Content-Type: application/json" --data '{"username":"admin","password":"'$ARGOCD_SECRET'"}' https://localhost:8443/api/v1/session | jq -r .token)
+
+# Create an API token to be used by the provider
+ARGOCD_PROVIDER_USER="provider-argocd"
+ARGOCD_TOKEN=$(curl -s -X POST -k -H "Authorization: Bearer $ARGOCD_SESSION_TOKEN" -H "Content-Type: application/json" --data '{"username":"admin","password":"'$ARGOCD_SECRET'"}' https://localhost:8443/api/v1/account/$ARGOCD_PROVIDER_USER/token | jq -r .token)
```
[the remark this answers is not in the document]
Comment (MisterMX) [reply README.md]: No it's a copy-paste mistake. I removed it.

File: README.md [hunk 2]
```diff
@@ -28,8 +28,12 @@ Install the [kubectl-view-secret](https://github.com/elsesiy/kubectl-view-secret
 ARGOCD_SECRET=$(kubectl view-secret argocd-initial-admin-secret -n argocd -q)
 # port forward the argocd api to the host:
 kubectl -n argocd port-forward svc/argocd-server 8443:443
-# create a JWT for the admin user at the argocd api
-ARGOCD_TOKEN=$(curl -s -X POST -k -H "Content-Type: application/json" --data '{"username":"admin","password":"'$ARGOCD_SECRET'"}' https://localhost:8443/api/v1/session | jq -r .token)
+# create a session JWT for the admin user at the argocd api
+ARGOCD_SESSION_TOKEN=$(curl -s -X POST -k -H "Content-Type: application/json" --data '{"username":"admin","password":"'$ARGOCD_SECRET'"}' https://localhost:8443/api/v1/session | jq -r .token)
+
+# Create an API token to be used by the provider
```
[the remark this answers is not in the document]
Comment (MisterMX) [reply README.md]: Good idea. I changed the description.

File: README.md [hunk 3]
```diff
@@ -28,8 +28,12 @@ Install the [kubectl-view-secret](https://github.com/elsesiy/kubectl-view-secret
 ARGOCD_SECRET=$(kubectl view-secret argocd-initial-admin-secret -n argocd -q)
 # port forward the argocd api to the host:
 kubectl -n argocd port-forward svc/argocd-server 8443:443
```
[the remark this answers is not in the document]
Comment (MisterMX) [reply README.md:45]: I added an extra step with a link to the official documentation on how to create new user accounts.

**Comment by janwillies** [comment]
lgtm

Thanks Max!

**Pull request merged**